src/bit-systems/network-receive-system.ts

```diff
@@ -71,26 +107,48 @@ export function networkReceiveSystem(world: HubsWorld) {
       const nid = APP.getSid(nidString);
 
       if (world.deletedNids.has(nid)) {
-        // TODO we may need to allow this for reconnects
         console.log(`Received a create message for an entity I've already deleted. Skipping ${nidString}`);
       } else if (world.nid2eid.has(nid)) {
         console.log(`Received create message for entity I already created. Skipping ${nidString}`);
       } else if (!hasPermissionToSpawn(creator, prefabName)) {
         // this should only ever happen if there is a bug or the sender is maliciously modified
-        console.log(`Received create from a user who does not have permission to spawn ${prefabName}`);
+        console.warn(`Received create from a user who does not have permission to spawn ${prefabName}`);
         world.ignoredNids.add(nid); // TODO should we just use deletedNids for this?
       } else {
-        const eid = createNetworkedEntityFromRemote(world, prefabName, initialData, nidString, creator, creator);
+        // TODO: Do we need to delete from ignoredNids?
+        // If permissions have changed, we don't want to permantently ignore messages.
+        // world.ignoredNids.delete(nid);
+
+        const eid = renderAsNetworkedEntity(world, prefabName, initialData, nidString, creator);
         console.log("got create message for", nidString, eid);
+      }
+    }
+  }
+
+  // If we stored updates for newly created entities, queue them for processing
+  enteredNetworkedQuery(world).forEach(eid => {
+    const nid = Networked.id[eid];
+    if (storedUpdates.has(nid)) {
+      console.log("Had stored updates for", APP.getString(nid), storedUpdates.get(nid));
+      const updates = storedUpdates.get(nid)!;
 
-        // If we were hanging onto updates for this nid we can now apply them. And they should be processed before other updates.
-        if (storedUpdates.has(nid)) {
-          console.log("had pending updates for", nidString, storedUpdates.get(nid));
-          Array.prototype.unshift.apply(message.updates, storedUpdates.get(nid));
-          storedUpdates.delete(nid);
+      for (let i = 0; i < updates.length; i++) {
+        const update = updates[i];
+        if (partedClientIds.has(APP.getSid(update.owner))) {
+          console.log("Rewriting update message from client who left.", JSON.stringify(update));
```

Comment:
This behavior seems suspicious.

We wanted to use the `timestamp` of the message as a way to ensure that whoever received the latest information from a previous owner takes ownership, but that could cause valid ownership transfers to be "undone." 

I would be more comfortable if this client used the `lastOwnerTime` or `lastOwnerTime + 1` of the message instead, so that any in-flight messages that we're about to receive take precedence over this stored message.

If we want to ensure that messages with the highest timestamp take precedence over those with lower timestamps, maybe that should be a condition in `isMessageOutdated`: First, check the `lastOwnerTime`. If they're the same, check the `timestamp`. If they're the same, then tiebreak on clientIds.

Reply:
Edit: After discussing, we've settled on using `timestamp` most of the time, not `lastOwnerTime`. We expect there to be _rare_ cases when this results in an unexpected ownership transfer, such as when someone takes ownership of an object just before the previous owner disconnects.